test/test-functions

```diff
@@ -485,19 +485,27 @@ check_asan_reports() {
             ret=$(($ret+1))
         fi
 
-	journald_report=$(find "$root" -name "systemd-journald.asan.log*" -exec cat {} \;)
-	if [[ ! -z "$journald_report" ]]; then
+        journald_report=$(find "$root" -name "systemd-journald.asan.log*" -exec cat {} \;)
+        if [[ ! -z "$journald_report" ]]; then
             printf "%s" "$journald_report"
             ret=$(($ret+1))
-	fi
+        fi
 
-	pids=$("$BUILD_DIR/journalctl" -D "$root/var/log/journal" | perl -alne 'print $1 if /\[(\d+)\]:\s*SUMMARY:\s+\w+Sanitizer/')
-	if [[ ! -z "$pids" ]]; then
+        pids=$(
+            "$BUILD_DIR/journalctl" -D "$root/var/log/journal" | perl -alne '
+                 BEGIN {
+                     %services_to_ignore = (
+                         "dbus-daemon" => undef,
+                     );
+                 }
+	         print $2 if /\s(\S*)\[(\d+)\]:\s*SUMMARY:\s+\w+Sanitizer/ && !exists $services_to_ignore{$1}'
```

Comment:
How about ignoring only "Leak" for dbus-daemon?

```perl
%services_to_ignore = (
    "dbus-daemon" => ["Leak"],
);
print $2 if /\s(\S*)\[(\d+)\]:\s*SUMMARY:\s+(\w+)Sanitizer/ && !($3 ~~ $services_to_ignore{$1});'
```

Reply:
BTW, `~~` is the [smart match operator](https://perldoc.perl.org/perlop.html#Smartmatch-Operator) available from Perl 5.10+... I hope that's alright.

Reply:
@filbranden thank you for the review!

I decided to ignore `dbus-daemon` altogether because although it's useful to catch issues there and report them to the maintainers of `dbus`, the test itself is supposed to test `systemd`. Some day `dbus` will be updated and ASan+UBsan will find something else and the test should be fixed relatively fast. I'm not sure it's likely to happen when the day comes :-) (and this PR is supposed to prevent the total breakage in the future).

Regarding `~~`, I kind of like it (even though it scares me sometimes :-)) I think `exists` is easier to understand, which is important given that perl fans are rare here :-)